src/app/Scenes/HomeView/Sections/HomeViewSectionViewingRooms.tsx

```diff
@@ -40,36 +40,24 @@ export const HomeViewSectionViewingRooms: React.FC<HomeViewSectionViewingRoomsPr
   const section = useFragment(viewingRoomsFragment, sectionProp)
   const viewAll = section.component?.behaviors?.viewAll
 
-  const onSectionViewAll = () => {
+  const onHeaderPress = () => {
     if (viewAll?.href) {
       tracking.tappedViewingRoomGroupViewAll(
         section.contextModule as ContextModule,
         viewAll?.ownerType as ScreenOwnerType
       )
-
-      navigate(viewAll.href)
-    } else {
-      tracking.tappedViewingRoomGroupViewAll(
-        section.contextModule as ContextModule,
-        section.ownerType as ScreenOwnerType
-      )
-
-      navigate(`/home-view/sections/${section.internalID}`, {
-        passProps: {
-          sectionType: section.__typename,
-        },
-      })
     }
   }
 
   return (
     <Flex {...flexProps}>
-      <Flex px={2}>
-        <SectionTitle
-          title={section.component?.title}
-          onPress={viewAll ? onSectionViewAll : undefined}
-        />
-      </Flex>
+      <SectionTitle
+        href={viewAll?.href}
```

Comment:
```suggestion
        href={viewAll?.href ? viewAll?.href : `/home-view/sections/${section.internalID}?sectionType=section.__typename`
```

This is the first case where I feel more customisation on `onPress` makes sense or an additional prop for `passProps`

Reply:
I've simplified this, which works for the navigation and prefetching. There is an additional prop `navigationProps` to pass the `passProps`, but having the params in the URL works as well

```
const href =
    viewAll?.href || `home-view/sections/${section.internalID}?sectionType=${section.__typename}`
```

Reply:
And moved `viewAll?.href || `home-view/sections/${section.internalID}?sectionType=${section.__typename}` into a helper function: `getHomeScreenHref`.